pkg/model/resource/resource.go

```diff
@@ -19,54 +19,113 @@ package resource
 
-	// ImportAlias is a cleaned concatenation of Group and Version.
-	ImportAlias string `json:"-"`
+	// Path is the path to the go package where the types are defined.
+	Path string `json:"path,omitempty"`
 
-	// Package is the go package of the Resource.
-	Package string `json:"package,omitempty"`
+	// API holds the information related to the resource API.
+	API *API `json:"api,omitempty"`
 
-	// Domain is the Group + "." + Domain of the Resource.
-	Domain string `json:"domain,omitempty"`
+	// Controller specifies if a controller has been scaffolded.
+	Controller bool `json:"controller,omitempty"`
 
-	// Namespaced is true if the resource is namespaced.
-	Namespaced bool `json:"namespaced,omitempty"`
+	// Webhooks holds the information related to the associated webhooks.
+	Webhooks *Webhooks `json:"webhooks,omitempty"`
+}
+
+// QualifiedGroup returns the fully qualified group name with the available information.
+func (r Resource) QualifiedGroup() string {
+	switch "" {
+	case r.Domain:
+		return r.Group
+	case r.Group:
+		return r.Domain
+	default:
+		return fmt.Sprintf("%s.%s", r.Group, r.Domain)
+	}
+}
+
+// PackageName returns a name valid to be used por go packages.
+func (r Resource) PackageName() string {
+	if r.Group == "" {
+		return safeImport(r.Domain)
+	}
+
+	return safeImport(r.Group)
+}
 
-	// CRDVersion holds the CustomResourceDefinition API version used for the Resource.
-	CRDVersion string `json:"crdVersion,omitempty"`
-	// WebhookVersion holds the {Validating,Mutating}WebhookConfiguration API version used for the Resource.
-	WebhookVersion string `json:"webhookVersion,omitempty"`
+// ImportAlias returns a identifier usable as an import alias for this resource.
+func (r Resource) ImportAlias() string {
+	if r.Group == "" {
+		return safeImport(r.Domain + r.Version)
+	}
+
+	return safeImport(r.Group + r.Version)
+}
+
+// GVK returns the GVK identifier of a resource.
+func (r Resource) GVK() GVK {
+	return GVK{
+		Group:   r.QualifiedGroup(),
+		Version: r.Version,
+		Kind:    r.Kind,
+	}
 }
 
-// GVK returns the group-version-kind information to check against tracked resources in the configuration file
-func (r *Resource) GVK() config.GVK {
-	return config.GVK{
-		Group:          r.Group,
-		Version:        r.Version,
-		Kind:           r.Kind,
-		CRDVersion:     r.CRDVersion,
-		WebhookVersion: r.WebhookVersion,
+// Update combines fields of two resources that have matching GVK favoring the receiver's values.
+func (r *Resource) Update(other *Resource) error {
+	// If other is nil, nothing to merge
+	if other == nil {
+		return nil
+	}
+
+	// If self is nil, set to other
+	if r == nil {
+		return fmt.Errorf("unable to update a nil Resource")
+	}
+
+	// Make sure we are not merging resources for different GVKs.
+	if !r.GVK().IsEqualTo(other.GVK()) {
+		return fmt.Errorf("unable to update a Resource with another with non-matching GVK")
+	}
+
+	// TODO: check that Plural & Path match, as we should never update these fields
```

Comment:
Should not it get done before the merge? Why this todo should be lived here?

Reply:
Right now I dont think you can mess with this as a user. But, for example, when plural is completely supported, if a create api and a create webhook calls are called wwith conflicting plurals, we should decide how to merge, or what error to yield.

Reply:
Wdyt about clarifies its comment and add more context to help others maintainer understand further why it is in the code?